python/cuml/naive_bayes/naive_bayes.py

```diff
@@ -0,0 +1,533 @@
+            n_features is the number of features
+
+        y : array-like of shape (n_samples) Target values.
+        classes : array-like of shape (n_classes)
+                  List of all the classes that can possibly appear in the y
+                  vector. Must be provided at the first call to partial_fit,
+                  can be omitted in subsequent calls.
+
+        sample_weight : array-like of shape (n_samples)
+                        Weights applied to individual samples (1. for
+                        unweighted). Currently sample weight is ignored
+
+        Returns
+        -------
+
+        self : object
+        """
+        return self._partial_fit(X, y, sample_weight=sample_weight,
+                                 _classes=classes)
+
+    @cp.prof.TimeRangeDecorator(message="predict()", color_id=1)
+    def predict(self, X):
+
+        """
+        Perform classification on an array of test vectors X.
+
+        Parameters
+        ----------
+
+        X : array-like of shape (n_samples, n_features)
+
+        Returns
+        -------
+
+        C : cupy.ndarray of shape (n_samples)
+
+        """
+
+        jll = self._joint_log_likelihood(X)
+        indices = cp.argmax(jll, axis=1).astype(self.classes_.dtype)
+
+        y_hat = invert_labels(indices, classes=self.classes_)
+        return y_hat
+
+    def predict_log_proba(self, X):
+
+        """
+        Return log-probability estimates for the test vector X.
+
+        Parameters
+        ----------
+
+        X : array-like of shape (n_samples, n_features)
+
+
+        Returns
+        -------
+
+        C : array-like of shape (n_samples, n_classes)
+            Returns the log-probability of the samples for each class in the
+            model. The columns correspond to the classes in sorted order, as
+            they appear in the attribute classes_.
+        """
+        jll = self._joint_log_likelihood(X)
+
+        # normalize by P(X) = P(f_1, ..., f_n)
+
+        # Compute log(sum(exp()))
+
+        # Subtract max in exp to prevent inf
+        a_max = cp.amax(jll, axis=1, keepdims=True)
+
+        logsumexp = cp.log(cp.sum(cp.exp(jll - a_max), axis=1))
+
+        a_max = cp.squeeze(a_max, axis=1)
+
+        log_prob_x = a_max + logsumexp
+
+        if log_prob_x.ndim < 2:
+            log_prob_x = log_prob_x.reshape((1, log_prob_x.shape[0]))
+        return jll - log_prob_x.T
+
+    def predict_proba(self, X):
+        """
+        Return probability estimates for the test vector X.
+
+        Parameters
+        ----------
+
+        X : array-like of shape (n_samples, n_features)
+
+        Returns
+        -------
+
+        C : array-like of shape (n_samples, n_classes)
+            Returns the probability of the samples for each class in the model.
+            The columns correspond to the classes in sorted order, as they
+            appear in the attribute classes_.
+        """
+        return cp.exp(self.predict_log_proba(X))
```

Comment:
rmm_cupy_ary

Reply:
I'll have to check this, but I believe this is returning a float on host.